textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -1,167 +1,222 @@
-    @Override
-    public void close() throws DataFlowException {
-        try {
-            operator.close();
-        } catch (Exception e) {
-            e.printStackTrace();
-            throw new DataFlowException(e.getMessage(), e);
-        }
-    }
-}
+
+package edu.uci.ics.textdb.dataflow.dictionarymatcher;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IDictionary;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.common.constants.SchemaConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.field.DataTuple;
+import edu.uci.ics.textdb.common.field.ListField;
+import edu.uci.ics.textdb.common.field.Span;
+import edu.uci.ics.textdb.common.field.StringField;
+import edu.uci.ics.textdb.common.field.TextField;
+
+/**
+ * @author Sudeep [inkudo]
+ * 
+ */
+public class DictionaryMatcher implements IOperator {
+
+    private IOperator operator;
+    private IDictionary dictionary;
+    private String dictionaryValue;
+    private String dictionaryValueDup;
+    private int positionIndex; // next position in the field to be checked.
+    private int attributeIndex; // Index of the next field to be checked.
+    private int spanIndexValue; // Starting position of the matched dictionary
+                                // string
+    private String spanFieldName;
+    private ITuple dataTuple;
+    private List<IField> fields;
+    private Schema schema;
+    private Schema spanSchema;
+
+    private String regex;
+    private Pattern pattern;
+    private Matcher matcher;
+    private List<Attribute> searchInAttributes;
+    private List<Span> spanList;
+    private boolean isPresent;
+
+    /**
+     * 
+     * @param dictionary
+     * @param operator
+     * @param searchInAttributes
+     *            --> The list of attributes to be searched in the tuple
+     */
+    public DictionaryMatcher(IDictionary dictionary, IOperator operator, List<Attribute> searchInAttributes) {
+        this.operator = operator;
+        this.dictionary = dictionary;
+        this.searchInAttributes = searchInAttributes;
+    }
+
+    /**
+     * @about Opens dictionary matcher. Initializes positionIndex and fieldIndex
+     *        Gets first sourceoperator tuple and dictionary value.
+     */
+    @Override
+    public void open() throws Exception {
+        try {
+            positionIndex = 0;
+            attributeIndex = 0;
+            operator.open();
+            dictionaryValue = dictionary.getNextValue();
+            dictionaryValueDup = dictionaryValue;
+            regex = "\\b" + dictionaryValueDup.toLowerCase() + "\\b";
+            pattern = Pattern.compile(regex);

```

Comment:
Add a comment here about how we check word boundaries (using regex). This logic may not be consistent with the analyzer logic. Any way to make them consistent?